plugins/prometheus/pkg/data/data.go

```diff
@@ -212,3 +268,119 @@ func getOwner(sample *model.SampleStream) Owner {
 		Container:           string(sample.Metric["container"]),
 	}
 }
+
+// adjustMetricsForMultiContainerPods splits values for any metrics
+// whos pod has more than one container. The original values
+// will be divided by the number of containers, with any remainder assigned to
+// the first container.
+//
+// For example, a pod with 3 containers, and a
+// metric value of 10, will be converted to 3 metrics. The metric for the
+// first container will have value 4, and the second and third metrics will
+// each have a value of 3.
+// A metric value of 1.5, will be converted to 3 metrics. The metric for the
+// first container will have value 1.5, and the second and third metrics will
+// each have a value of 0.
+//
+// The specified map of Kubernetes workloads is used to determine the number of
+// containers and their names, for pods referenced in each metric. The map
+// should be
+// keyed on a string `{namespace}/{podName}`.
+func adjustMetricsForMultiContainerPods(metrics model.Matrix, workloadMap map[string]*controller.Workload) model.Matrix {
```

Comment:
This function is a bit long, however I didn't break it out into smaller functions because doing so requires me to pass more values in-and-out of smaller functions, trading this long function for the complexity of more setup and value-passing.